x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/components/source_added.tsx

```diff
@@ -4,52 +4,30 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React from 'react';
+import React, { useEffect } from 'react';
 
 import { Location } from 'history';
-import { useActions, useValues } from 'kea';
-import { Redirect, useLocation } from 'react-router-dom';
+import { useActions } from 'kea';
+import { useLocation } from 'react-router-dom';
 
-import { i18n } from '@kbn/i18n';
+import { parseQueryParams } from '../../../../shared/query_params';
+import { Loading } from '../../../../shared/loading';
 
-import { setErrorMessage } from '../../../../shared/flash_messages';
-
-import { parseQueryParams } from '../../../../../applications/shared/query_params';
-
-import { SOURCES_PATH, getSourcesPath } from '../../../routes';
-
-import { AppLogic } from '../../../app_logic';
-import { SourcesLogic } from '../sources_logic';
-
-interface SourceQueryParams {
-  name: string;
-  hasError: boolean;
-  errorMessages?: string[];
-  serviceType: string;
-  indexPermissions: boolean;
-}
+import { AddSourceLogic, OauthParams } from './add_source/add_source_logic';
 
+/**
+ * This component merely triggers catchs the redirect from the oauth application and initializes the saving
+ * of the params the oauth plugin sends back. The logic file now redirects back to sources with either a
+ * success or error message upon completion.
+ */
 export const SourceAdded: React.FC = () => {
   const { search } = useLocation() as Location;
-  const { name, hasError, errorMessages, serviceType, indexPermissions } = (parseQueryParams(
-    search
-  ) as unknown) as SourceQueryParams;
-  const { setAddedSource } = useActions(SourcesLogic);
-  const { isOrganization } = useValues(AppLogic);
-  const decodedName = decodeURIComponent(name);
+  const params = (parseQueryParams(search) as unknown) as OauthParams;
```

Comment:
Could we put these two lines into the logic file as well? So that this component only triggers the action and not parses the URL. I think it could simplify tests as well.

Reply:
Which 2 lines exactly? We don't have access to Hooks in Logic files so we can't use `useLocation`

Reply:
I moved the line you commented on [here](https://github.com/elastic/kibana/pull/90067/commits/7dee6cd21e4c74428b86322f4927b8367bac6228)